tests/flow/test_v7_encode_decode.py

```diff
@@ -0,0 +1,137 @@
+        # Create 3 nodes meta keys
+        redis_graph.query("UNWIND range(0,20) as i CREATE ()")
+        # Return all the nodes, before and after saving & loading the RDB, and check equality
+        query = "MATCH (n) WITH n ORDER by id(n) return COLLECT(id(n))"
+        expected_full_graph_nodes_id = redis_graph.query(query)
+        # Delete 3 nodes.
+        redis_graph.query("MATCH (n) WHERE id(n) IN [7,14,20] DELETE n")
+        expected_nodes_id_after_delete = redis_graph.query(query)
+        # Save RDB & Load from RDB
+        redis_con.execute_command("DEBUG", "RELOAD")
+        actual = redis_graph.query(query)
+        # Validate no compaction, all IDs are the same
+        self.env.assertEquals(expected_nodes_id_after_delete.result_set, actual.result_set)
+        # Validate reuse of node ids - create 3 nodes.
+        redis_graph.query("UNWIND range (0,2) as i CREATE ()")
+        actual = redis_graph.query(query)
+        self.env.assertEquals(expected_full_graph_nodes_id.result_set, actual.result_set)
+
+    def test03_edges_over_multiple_keys(self):
+        graph_name = "edges_over_multiple_keys"
+        redis_graph = Graph(graph_name, redis_con)
+        # Create 3 edges meta keys
+        redis_graph.query("UNWIND range(0,20) as i CREATE ()-[:R {val:i}]->()")
+        # Return all the edges, before and after saving & loading the RDB, and check equality
+        query = "MATCH ()-[e]->() return e"
+        expected = redis_graph.query(query)
+        # Save RDB & Load from RDB
+        redis_con.execute_command("DEBUG", "RELOAD")
+        actual = redis_graph.query(query)
+        self.env.assertEquals(expected.result_set, actual.result_set)
+
+    def test04_no_compaction_on_edges_delete(self):
+        graph_name = "no_compaction_on_edges_delete"
+        redis_graph = Graph(graph_name, redis_con)
+        # Create 3 nodes meta keys
+        redis_graph.query("UNWIND range(0,20) as i CREATE ()-[:R]->()")
+        # Return all the edges, before and after saving & loading the RDB, and check equality
+        query = "MATCH ()-[e]->() WITH e ORDER by id(e) return COLLECT(id(e))"
+        expected_full_graph_nodes_id = redis_graph.query(query)
+        # Delete 3 edges.
+        redis_graph.query("MATCH ()-[e]->() WHERE id(e) IN [7,14,20] DELETE e")
+        expected_nodes_id_after_delete = redis_graph.query(query)
+        # Save RDB & Load from RDB
+        redis_con.execute_command("DEBUG", "RELOAD")
+        actual = redis_graph.query(query)
+        # Validate no compaction, all IDs are the same
+        self.env.assertEquals(expected_nodes_id_after_delete.result_set, actual.result_set)
+        # Validate reuse of edges ids - create 3 edges.
+        redis_graph.query("UNWIND range (0,2) as i CREATE ()-[:R]->()")
+        actual = redis_graph.query(query)
+        self.env.assertEquals(expected_full_graph_nodes_id.result_set, actual.result_set)
+
+    def test05_multiple_edges_over_multiple_keys(self):
```

Comment:
How is this test different from `test03_edges_over_multiple_keys` ?

Reply:
multiple edges and simple edges have different decode mechanism. I want to validate the data integrity